src/Product/FilteredProductList.php

```diff
@@ -0,0 +1,90 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Product;
+
+use Countable;
+use WC_Product;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class FilteredProductList
+ *
+ * A list of filtered products and their total count before filtering.
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product
+ */
+class FilteredProductList implements Countable {
+
+	/**
+	 * List of product objects or IDs.
+	 *
+	 * @var WC_Product[]|int[]
+	 */
+	protected $products;
+
+	/**
+	 * Count before filtering.
+	 *
+	 * @var int
+	 */
+	protected $unfiltered_count;
+
+	/**
+	 * FilteredProductList constructor.
+	 *
+	 * @param array $products         List of filtered products.
```

Comment:
probably best to have the types here too

```suggestion
	 * @param WC_Product[]|int[] $products         List of filtered products.
```